actions/treasures.go

```diff
@@ -15,7 +24,51 @@ type treasureReq struct {
 	EthKey          string `json:"EthKey"`
 }
 
+type treasureRes struct {
+	Success string `json:"success"`
+}
+
+func unixMilli(t time.Time) int64 {
+	return t.Round(time.Millisecond).UnixNano() / (int64(time.Millisecond) / int64(time.Nanosecond))
+}
+
 // Verifies the treasure and claims such treasure.
 func (t *TreasuresResource) VerifyAndClaim(c buffalo.Context) error {
-	return nil
-}
+	req := treasureReq{}
+	oyster_utils.ParseReqBody(c.Request(), &req)
+
+	host_ip := os.Getenv("HOST_IP")
+	provider := "http://" + host_ip + ":14265"
+	api := giota.NewAPI(provider, nil)
+	ftr := &giota.FindTransactionsRequest{Bundles: []giota.Trytes{"DEXRPLKGBROUQMKCLMRPG9HFKCACDZ9AB9HOJQWERTYWERJNOYLW9PKLOGDUPC9DLGSUH9UHSKJOASJRU"}}
+	resp, err := api.FindTransactions(ftr)
+
+	datamap1, vErr := models.GetDataMap(req.GenesisHash, req.NumChunks)
+	for _, d := range datamap1 {
+		
+	}
+	// or 
+	
+	vErr, error := models.BuildDataMaps(req.GenesisHash, req.NumChunks)
```

Comment:
Calling this method will actually insert the datamap into the brokernode's DB table.

Reply:
Yes, so where to save datamap? To database or on fly or dont save? If into database, should I create new table or is something prepared?

Reply:
There is currently no table yet for saving data maps that the brokernode is currently in the process of doing PoW for.  So, if we decide to go that route, you'll have to create one.  

There's some decisions we have to make.  We probably don't want the brokernode to have to verify that ALL transactions in the sector had PoW done on them within the last epoch, since that could be up to 1 million verifications.  What I think we should do is have the brokernode randomly verify that X transactions from the sector have had the PoW done in the last epoch, where X is some number that's large enough that the webnode will not feel comfortable cheating since it will likely get caught.  

I would recommend we build this in such a way that X is easily changed, and divide the sector to be verified into X subsectors.  Each subsector would have a length of 1,000,000 / X.  Randomly pick a single transaction from within each subsector to verify.  That way we aren't re-verifying any transactions we've already verified and we know that our verifications are evenly distributed throughout the datamap.  

Whether it makes sense to save this to a DB table or do it on the fly really depends on how large X is.  If X is fairly small, doing it on the fly would be fine.  If X is fairly large, I would think we'd want it in a table.  

I'm leaning towards suggesting we need tables for this.  Doing it in a table would still work, even if X is small, whereas doing it on the fly might not work so well if X is large, so with a table we've got both our bases covered.  You'd probably want 2 tables.  One named something like "verification_sessions" which would store the values sent to it by the webnode and anything else it needs, including the payout ETH address, and then another called something like "verifiable_data_maps" that would store the session ID of the verification request and the actual addresses and the timestamps of the transactions.  

With mainnet coming up soon though, we could always just keep X fairly small, and do the checking on the fly, and then implement a change to move it to a table after mainnet for more thorough verification.  

@AaronVasquez    thoughts?

Reply:
Agree with everything you just said. Lets go with on the fly for this PR, and work on optimizing and fault tolerance later. Let's start with verifying 5 random transactions? This number is probably too small, but we can tweak this in the future.

Reply:
Yeah we'll probably want to do at least like 1000 or so for the launch, but as long as we're saving it in a variable somewhere that's easily changeable, that's what matters.  

@tenisakb For testing purposes, it would be nice if you could write this functionality in such a way that it's easily testable.  I.e. we should be able to verify that if we tell it to verify 5 random transactions, the indexes it will return will all be somewhere in the subsectors.  I.e. the first random transaction is between 0 and 199,000, etc, the second randomly selected value is between 200,000 and 399,999, etc.

Reply:
You could have a separate method that picks these random indexes and just returns them.